docs(rust_plugin): update explanations to reflect changes in newer Rust toolchains (#850)
canonical/craft-parts: docs/common/craft-parts/reference/plugins/rust_plugin.rst

* (Advanced) Perform cross-language LTO. This requires installing the correct version of LLVM/Clang and setting the right environment variables.
  You must know which LLVM version of your selected Rust toolchain is using.

>   You can use ``rustc -vV`` to print out the LLVM version used by the compiler. For example, you can see Rust 1.81 uses LLVM 18.1 because
>   it prints out an output like this:

Reviewer: ```suggestion
  You can use ``rustc -vV`` to check the LLVM version used by the compiler. For example, you can see Rust 1.81 uses LLVM 18.1 because
  it prints an output like this:
```

  .. code-block:: text
    rustc 1.81.0 (eeb90cda1 2024-09-04)
    binary: rustc
    commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
    commit-date: 2024-09-04
    host: x86_64-unknown-linux-gnu
    release: 1.81.0
    LLVM version: 18.1.7

>   On Rust toolchains that do not include the LLVM version, you can still check the LLVM version number by examining the ``lib`` directory.

Reviewer: ```suggestion
  On Rust toolchains that don't include the LLVM version, you can check the LLVM version number by examining the ``lib`` directory.
```

>   For example, Rust 1.81 uses LLVM 18.1 because it bundles a ``libLLVM.so.18.1-rust-1.81.0-stable`` file under the ``lib`` directory.
> 
>   In this case, you want to install ``clang-18`` and ``lld-18`` from the Ubuntu archive.

Reviewer: ```suggestion
  For example, Rust 1.81 uses LLVM 18.1 because it bundles a ``libLLVM.so.18.1-rust-1.81.0-stable`` file under the ``lib`` directory.
  In this case, you would install ``clang-18`` and ``lld-18`` from the Ubuntu archive.
```

* (Advanced) Perform cross-language LTO. This requires installing the correct version of LLVM/Clang and setting the right environment variables.
  You must know which LLVM version of your selected Rust toolchain is using.

>   For example, Rust 1.81 uses LLVM 18 because you can see it bundles a ``libLLVM.so.18.1-rust-1.81.0-stable`` file in the ``lib`` directory.

Reviewer 2: Is there a document somewhere that lists the correct LLVM versions for rust versions?

Reviewer: I couldn't find anything definitive in a cursory search. Their doc on upgrading LLVM points to GH PRs, so they might be the most reliable way to check. Said PRs at least have  consistent naming: https://github.com/rust-lang/rust/pulls?q=is%3Apr+is%3Aclosed+in%3Atitle+%22Update+to+LLVM%22+

Author: > Is there a document somewhere that lists the correct LLVM versions for rust versions?

You might also do `rustc -vV` on newer Rust versions, which will print out something like:

```
rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-unknown-linux-gnu
release: 1.81.0
LLVM version: 18.1.7
```

Reviewer 2: @liushuyu could you include that tip in the doc for users?
